File: 2150-find-all-lonely-numbers-in-the-array/2150-find-all-lonely-numbers-in-the-array.cpp
class Solution {
public:
    vector<int> findLonely(vector<int>& nums) {
        map<int,int>mp;
        vector<int>ans;
        for(int i:nums){
            mp[i]++;
        }
        for(auto i:mp){
            if(i.second==1){
                if(mp.find(i.first+1)==mp.end() && mp.find(i.first-1)==mp.end()){
                  ans.push_back(i.first);
                }
            }
        }
        return ans;
    }
};

File: 2150-find-all-lonely-numbers-in-the-array/2150-find-all-lonely-numbers-in-the-array.java
class Solution {
    public List<Integer> findLonely(int[] nums) {
        HashMap<Integer,Integer>mp=new HashMap<>();
        ArrayList<Integer>ans=new ArrayList<>();
        for(int i:nums){
            mp.put(i,mp.getOrDefault(i,0)+1);
        }
          for(Map.Entry<Integer, Integer>entry:mp.entrySet()) {
            Integer key=entry.getKey();
            Integer value=entry.getValue();
            if(value==1){
                   if (!mp.containsKey(key+1) && !mp.containsKey(key-1)){
                    ans.add(key);
                }
            }
          }
          return ans;
    }
}